cmd/frontend/internal/highlight/filetype.go

```diff
@@ -0,0 +1,165 @@
+package highlight
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-enry/go-enry/v2"
+	"github.com/grafana/regexp"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+)
+
+type EngineType int64
+
+const (
+	EngineInvalid    EngineType = 0
+	EngineTreeSitter            = 1
+	EngineSyntect               = 2
+)
+
+type ftPattern struct {
```

Comment:
Took me a while to guess what `ft` referred to
```suggestion
type filetypePattern struct {
```

Reply:
Given that enry has a `GetLanguage` method, should we use "language" instead of "filetype"?